node.go

```diff
@@ -104,26 +143,21 @@ func DecodeInto(b []byte, v interface{}) error {
 func decodeCBOR(b []byte) (m interface{}, err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			err = fmt.Errorf("cbor panic: %s", r)
+			err = fmt.Errorf("failed to unmarshal - cbor panic: %s", r)
 		}
 	}()
-	dec := cbor.NewDecoder(bytes.NewReader(b))
-	dec.TagDecoders[CBORTagLink] = new(IpldLinkDecoder)
-	err = dec.Decode(&m)
-	return
-}
 
-var ErrNoSuchLink = errors.New("no such link found")
-
-type Node struct {
-	obj   interface{}
-	tree  []string
-	links []*node.Link
-	raw   []byte
-	cid   *cid.Cid
+	err = cbor.UnmarshalAtlased(b, &m, cborAtlas)
+	if err != nil {
+		fmt.Printf("bytes: %s\n", hex.EncodeToString(b))
```

Comment:
btw, you can use `%x` instead of `%s` to automatically encode to hex.

Reply:
sweet, thank you <3